src/macros.js

```diff
@@ -38,6 +43,156 @@ defineMacro("\\iff", "\\;\\Longleftrightarrow\\;");
+    '\\xrightarrow': '\\dotsb',
+    '\\xleftarrow': '\\dotsb',
+    // Symbols whose definition starts with \DOTSI:
+    '\\DOTSI': '\\dotsi',
+    '\\int': '\\dotsi',
+    '\\oint': '\\dotsi',
+    '\\iint': '\\dotsi',
+    '\\iiint': '\\dotsi',
+    '\\iiiint': '\\dotsi',
+    '\\idotsint': '\\dotsi',
+    // Symbols whose definition starts with \DOTSX:
+    '\\DOTSX': '\\dotsx',
+};
+
+defineMacro("\\dots", function() {
+    // TODO: If used in text mode, should expand to \textellipsis.
+    // However, in KaTeX, \textellipsis and \ldots behave the same
+    // (in text mode), and it's unlikely we'd see any of the math commands
+    // that affect the behavior of \dots when in text mode.  So fine for now
+    // (until we support \ifmmode ... \else ... \fi).
+    let thedots = '\\dotso';
+    const next = this.expandAfterFuture().text;
+    if (next in dotsByToken) {
+        thedots = dotsByToken[next];
+    } else if (next.substr(0, 4) === '\\not') {
+        thedots = '\\dotsb';
+    } else if (next in symbols.math) {
+        if (utils.contains(['bin', 'rel'], symbols.math[next].group)) {
+            thedots = '\\dotsb';
+        }
+    }
+    return thedots;
+});
+
+const spaceAfterDots = {
+    // \rightdelim@ checks for the following:
+    ')': true,
+    ']': true,
+    '\\rbrack': true,
+    '\\}': true,
+    '\\rbrace': true,
+    '\\rangle': true,
+    '\\rceil': true,
+    '\\rfloor': true,
+    '\\rgroup': true,
+    '\\rmoustache': true,
+    '\\right': true,
+    '\\bigr': true,
+    '\\biggr': true,
+    '\\Bigr': true,
+    '\\Biggr': true,
+    // \extra@ also tests for the following:
+    '$': true,
+    // \extrap@ checks for the following:
+    ';': true,
+    '.': true,
+    ',': true,
+};
+
+defineMacro("\\dotso", function() {
+    const next = this.future().text;
+    if (next in spaceAfterDots) {
+        return "\\ldots\\,";
+    } else {
+        return "\\ldots";
+    }
+});
+
+defineMacro("\\dotsc", function() {
+    const next = this.future().text;
+    // \dotsc uses \extra@ but not \extrap@, instead specially checking for
+    // ';' and '.', but doesn't check for ','.
+    if (next in spaceAfterDots && next !== ',') {
+        return "\\ldots\\,";
+    } else {
+        return "\\ldots";
+    }
+});
+
+defineMacro("\\cdots", function() {
+    const next = this.future().text;
+    if (next in spaceAfterDots) {
+        return "\\@cdots\\,";
+    } else {
+        return "\\@cdots";
+    }
+});
+
+defineMacro("\\dotsb", "\\cdots");
+defineMacro("\\dotsm", "\\cdots");
+defineMacro("\\dotsi", "\\!\\cdots");
+// amsmath doesn't actually define \dotsx, but \dots followed by a macro
+// starting with \DOTSX implies \dotso, and then \extra@ detects this case
+// and forces the added `\,`.
+defineMacro("\\dotsx", "\\ldots\,");
+
+// \let\DOTSI\relax
+// \let\DOTSB\relax
+// \let\DOTSX\relax
+defineMacro("\\DOTSI", "\\relax");
+defineMacro("\\DOTSB", "\\relax");
+defineMacro("\\DOTSX", "\\relax");
```

Comment:
amsmath defines these "null macros" for other macros to indicate the behavior of `\dots` preceding them.  Indeed, I should have added use of these macros to `\iff`, `\implies`, and `\impliedby`.  See new commit.